### DISCUSSION TITLE
(Solved) Repository community implementing

### DISCUSSION THREAD

**Discussion by 01fiore** [mocaccinoOS/mocaccino#72] [General] [answered]
Hi, I need help implementing the repository community, merci.
https://packages.mocaccino.org/repository-index/repository/mocaccino-community/20211005/

**Comment by msdobrescu** [comment, marked as answer]
Hi, do you mean [this](https://github.com/mocaccinoOS/community-repository#readme)?

**Reply by mudler** [reply]
Note you can also just

`luet install repository/mocaccino-community`

To install it in your system

**Reply by msdobrescu** [reply]
The index is really outdated.

**Reply by mudler** [reply]
Are you sure? Looks good here https://github.com/mocaccinoOS/repository-index/blob/master/packages/mocaccino-community.yml

**Reply by 01fiore** [reply]
I confirm the problem with the repository:

moca /home/fiore # luet install repository/mocaccino-community
 INFO  Luet 0.32.0-g2aa4c8a42ed13682b4820f3ca5e6d2388a5e2c24 2022-05-03
 INFO  Copyright (C) 2019-2022 Ettore Di Giacinto
       This program comes with ABSOLUTELY NO WARRANTY.
       This is free software, and you are welcome to redistribute it under certain conditions.
       For documentation, visit https://luet.io.
 INFO   :::> Install
 INFO   🤔  Checking for available upgrades                                                                                                                                                                       
 INFO   Nothing to upgrade
 INFO   Packages that are going to be installed in the system:

Program Name                            | Version  | License | Repository                
repository/mocaccino-community-20211005 | 20211005 |         | mocaccino-repository-index

 INFO   By going forward, you are also accepting the licenses of the packages that you are going to install in your system.
 INFO   Do you want to continue with this operation? [y/N]: 
y
Downloading packages [0/1] █                                                                                                                                                                               0% | 1s
mocaccino-community-repository-20211005.package.tar.gz [1/329] █                                                                                                                                           0% | 0s
 INFO   Downloaded mocaccino-community-repository-20211005.package.tar.gz of 0.00 MB ( 0.00 MiB/s )

  ERROR    Failed downloading package mocaccino-community Artifact integrity check failure: Checksum mismsatch
 INFO   Checking for file conflicts..
  ERROR    Error: file conflict found: Failed downloading package: Artifact integrity check failure: Checksum mismsatch
panic: fatal error

goroutine 1 [running]:
github.com/mudler/luet/pkg/api/core/logger.(*Logger).Fatal(...)
        /luetbuild/luet/pkg/api/core/logger/logger.go:255
github.com/mudler/luet/cmd.glob..func10(0x162a8c0?, {0xc000147140?, 0x1?, 0x1?})
        /luetbuild/luet/cmd/install.go:94 +0x5c5
github.com/spf13/cobra.(*Command).execute(0x162a8c0, {0xc000147110, 0x1, 0x1})
        /luetbuild/luet/vendor/github.com/spf13/cobra/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x1628ac0)
        /luetbuild/luet/vendor/github.com/spf13/cobra/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /luetbuild/luet/vendor/github.com/spf13/cobra/command.go:902
github.com/mudler/luet/cmd.Execute()
        /luetbuild/luet/cmd/root.go:124 +0x25
main.main()
        /luetbuild/luet/main.go:10 +0x17

**Reply by msdobrescu** [reply]
> Are you sure? Looks good here https://github.com/mocaccinoOS/repository-index/blob/master/packages/mocaccino-community.yml

I mean here: https://packages.mocaccino.org/repository-index/repository/mocaccino-community/20211005/

**Reply by mudler** [reply]
> > Are you sure? Looks good here https://github.com/mocaccinoOS/repository-index/blob/master/packages/mocaccino-community.yml
> 
> I mean here: https://packages.mocaccino.org/repository-index/repository/mocaccino-community/20211005/

that looks quite behind yup, but has nothing to do with the repository-index. the package version gets bumped only if there is a repository definition change.

**Reply by 01fiore** [reply]
You can confirm that it is resolved: luet repo update && luet upgrade -y

# luet repo update
 INFO   Downloaded repository.yaml of 0.00 MB ( 0.00 MiB/s )
 INFO   Downloaded tree.tar.zst of 0.00 MB ( 0.01 MiB/s )
 INFO   Downloaded repository.meta.yaml.tar.zst of 0.00 MB ( 0.02 MiB/s )
 INFO   🏠  Repository luet-official revision: 1 (2022-05-03 19:32:54 +0200 CEST)
 INFO   ℹ  Repository: luet Priority: 1 Type: http
 INFO   Downloading quay.io/mocaccino/desktop:20220507191900-repository.yaml
 INFO   Pulled: sha256:3b579be05bb29aa426d93add01a08422533b66c7986c3163e31d1417f3948574
 INFO   Size: 679B
 INFO   Downloading quay.io/mocaccino/desktop:20220507191900-tree.tar.gz
 INFO   Pulled: sha256:776cb7bd48158f9cd04a5c5a55f20e7c8673f4802938ee72576a34f07d8494b1
 INFO   Size: 42.5KiB
 INFO   Downloading quay.io/mocaccino/desktop:20220507191900-repository.meta.yaml.tar
 INFO   Pulled: sha256:9e700940faf8f53e5a226b9e3e6eb4b054a8d19024e95857a9adad8490af0f7f
 INFO   Size: 47.54MiB
 INFO   🏠  Repository MocaccinoOS Desktop revision: 350 (2022-05-07 21:19:00 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-desktop-stable Priority: 3 Type: docker
 INFO   Downloading quay.io/mocaccino/extra:repository.yaml
 INFO   Pulled: sha256:dccc36b9a20c9ab00e0a702148f67899c9e3f93d73cd4d7426d79f2d238c70bf
 INFO   Size: 630B
 INFO   Downloading quay.io/mocaccino/extra:tree.tar.gz
 INFO   Pulled: sha256:c01e18bf3c241f7db8882a956a8fce6321f9053c166b5ecaf9ea0341d3b65922
 INFO   Size: 6.698KiB
 INFO   Downloading quay.io/mocaccino/extra:repository.meta.yaml.tar
 INFO   Pulled: sha256:71466f052e7f3f9b21df00995af862da350d7a08aa64561c537abe538ac388a5
 INFO   Size: 741KiB
 INFO   🏠  Repository mocaccino-extra revision: 107 (2022-05-07 19:59:57 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-extra-stable Priority: 2 Type: docker
 INFO   Downloading quay.io/mocaccino/os-commons:repository.yaml
 INFO   Pulled: sha256:bbeae9bbd78d6b393c703f1ba1f5ddb0259ddba5fba1c07e1b488adfe7585f09
 INFO   Size: 632B
 INFO   Downloading quay.io/mocaccino/os-commons:tree.tar.gz
 INFO   Pulled: sha256:01aace9544e9cc6dc38758faa6e429976386d91108a2c8e0fe46817fe481171f
 INFO   Size: 26.43KiB
 INFO   Downloading quay.io/mocaccino/os-commons:repository.meta.yaml.tar
 INFO   Pulled: sha256:576d349ee2cee8c0a5514eaefcec27faa16a063ec9d392d64e351a78996d1a5f
 INFO   Size: 967.5KiB
 INFO   🏠  Repository example-repository revision: 28 (2022-05-07 19:20:18 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-os-commons-stable Priority: 2 Type: docker
 INFO   Downloaded repository.yaml of 0.00 MB ( 0.00 MiB/s )
 INFO   Downloaded tree.tar.gz of 0.00 MB ( 0.01 MiB/s )
 INFO   Downloaded repository.meta.yaml.tar.gz of 0.00 MB ( 0.02 MiB/s )
 INFO   🏠  Repository mocaccino-repository-index revision: 1 (2022-05-08 10:51:11 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-repository-index Priority: 1 Type: http

**Comment by msdobrescu** [comment]
Would a full update/upgrade help?

`luet repo update && luet upgrade -y
`

**Reply by 01fiore** [reply]
Fixed with your command:
# luet repo update && luet upgrade -y 
 INFO   Downloaded repository.yaml of 0.00 MB ( 0.00 MiB/s )
 INFO   Downloaded tree.tar.zst of 0.00 MB ( 0.00 MiB/s )
 INFO   Downloaded repository.meta.yaml.tar.zst of 0.00 MB ( 0.01 MiB/s )
 INFO   🏠  Repository luet-official revision: 1 (2022-05-03 19:32:54 +0200 CEST)
 INFO   ℹ  Repository: luet Priority: 1 Type: http
 INFO   Downloading quay.io/mocaccino/desktop:20220503164425-repository.yaml
 INFO   Pulled: sha256:786046f60760f358401631333ba880d6a8f3da020262441fa34152a07cea84b1
 INFO   Size: 679B
 INFO   Downloading quay.io/mocaccino/desktop:20220503164425-tree.tar.gz
 INFO   Pulled: sha256:5915203ef675ac83fc80bd225458a6b58d72d0ed7fc4939b3b7e906f3b6f942e
 INFO   Size: 43.23KiB
 INFO   Downloading quay.io/mocaccino/desktop:20220503164425-repository.meta.yaml.tar
 INFO   Pulled: sha256:a27f61b04a1290ba5d795d2edff8590b3556b0a8f8f6b2e74e2a0f461b2992fe
 INFO   Size: 47.46MiB
 INFO   🏠  Repository MocaccinoOS Desktop revision: 348 (2022-05-03 18:44:25 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-desktop-stable Priority: 3 Type: docker
 INFO   Downloading quay.io/mocaccino/extra:repository.yaml
 INFO   Pulled: sha256:dccc36b9a20c9ab00e0a702148f67899c9e3f93d73cd4d7426d79f2d238c70bf
 INFO   Size: 630B
 INFO   Downloading quay.io/mocaccino/extra:tree.tar.gz
 INFO   Pulled: sha256:c01e18bf3c241f7db8882a956a8fce6321f9053c166b5ecaf9ea0341d3b65922
 INFO   Size: 6.698KiB
 INFO   Downloading quay.io/mocaccino/extra:repository.meta.yaml.tar
 INFO   Pulled: sha256:71466f052e7f3f9b21df00995af862da350d7a08aa64561c537abe538ac388a5
 INFO   Size: 741KiB
 INFO   🏠  Repository mocaccino-extra revision: 107 (2022-05-07 19:59:57 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-extra-stable Priority: 2 Type: docker
 INFO   Downloading quay.io/mocaccino/os-commons:repository.yaml
 INFO   Pulled: sha256:bbeae9bbd78d6b393c703f1ba1f5ddb0259ddba5fba1c07e1b488adfe7585f09
 INFO   Size: 632B
 INFO   Downloading quay.io/mocaccino/os-commons:tree.tar.gz
 INFO   Pulled: sha256:01aace9544e9cc6dc38758faa6e429976386d91108a2c8e0fe46817fe481171f
 INFO   Size: 26.43KiB
 INFO   Downloading quay.io/mocaccino/os-commons:repository.meta.yaml.tar
 INFO   Pulled: sha256:576d349ee2cee8c0a5514eaefcec27faa16a063ec9d392d64e351a78996d1a5f
 INFO   Size: 967.5KiB
 INFO   🏠  Repository example-repository revision: 28 (2022-05-07 19:20:18 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-os-commons-stable Priority: 2 Type: docker
 INFO   Downloaded repository.yaml of 0.00 MB ( 0.00 MiB/s )
 INFO   Downloaded tree.tar.gz of 0.00 MB ( 0.01 MiB/s )
 INFO   Downloaded repository.meta.yaml.tar.gz of 0.00 MB ( 0.01 MiB/s )
 INFO   🏠  Repository mocaccino-repository-index revision: 1 (2022-05-08 10:51:11 +0200 CEST)
 INFO   ℹ  Repository: mocaccino-repository-index Priority: 1 Type: http
 INFO  Luet 0.32.0-g2aa4c8a42ed13682b4820f3ca5e6d2388a5e2c24 2022-05-03
 INFO  Copyright (C) 2019-2022 Ettore Di Giacinto
       This program comes with ABSOLUTELY NO WARRANTY.
       This is free software, and you are welcome to redistribute it under certain conditions.
       For documentation, visit https://luet.io.
 INFO   :::> Upgrade
 INFO   🤔  Computing upgrade, please hang tight... 💤                                                                                                                                                            
 INFO   ⚡  Proposed version changes to the system:
        

Old version                                        | New version                                        | License
repository/mocaccino-desktop-stable-20220503164425 | repository/mocaccino-desktop-stable-20220507191900 |        
repo-updater/mocaccino-desktop-stable-0.1+18       | repo-updater/mocaccino-desktop-stable-0.1+19       |        


Downloading packages [0/2] █                                                                                                                                                                               0% | 0s
mocaccino-desktop-stable-repository-20220507191900.package.tar.gz [1/339] █                                                                                                                                0% | 0s
Downloading packages [0/2] █                                                                                                                                                                               0% | 0s
mocaccino-desktop-stable-repository-20220507191900.package.tar.gz [1/339] ████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████ 0% | 0s
mocaccino-desktop-stable-repo-updater-0.1+19.package.tar.gz [1/34] ████                                                                                                                                    3% | 0s
Downloading packages [1/2] ███████████████████████████████████████████████████████████████████████████████████████                                                                                        50% | 0s

Downloading packages [2/2] █████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████ 100% | 0s
mocaccino-desktop-stable-repository-20220507191900.package.tar.gz [339/339] ████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████ 100% | 0s
mocaccino-desktop-stable-repo-updater-0.1+19.package.tar.gz [34/34] ████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████ 100% | 0s
 INFO   Checking for file conflicts..
 INFO   Done checking for file conflicts..
 INFO   📦  Package  repo-updater/mocaccino-desktop-stable-0.1+19 installed
 INFO   Found protected file /etc/luet/repos.conf.d/mocaccino-desktop-stable.yml. Creating /etc/luet/repos.conf.d/._cfg0001_mocaccino-desktop-stable.yml.
 INFO   📦  Package  repository/mocaccino-desktop-stable-20220507191900 installed
 INFO   Executing finalizer for repo-updater/mocaccino-desktop-stable-0.1+19
 INFO   🐚  Executing finalizer on  / sh [-c mos config-update update --interactive=false -p /etc/luet/repos.conf.d || true]
 INFO   Unmerged configuration files: 1
       Latest changeset for /etc/luet/repos.conf.d/mocaccino-desktop-stable.yml (1)
       Merging configuration for file: /etc/luet/repos.conf.d/mocaccino-desktop-stable.yml (changeset /etc/luet/repos.conf.d/._cfg0001_mocaccino-desktop-stable.yml)
       ERROR: 0 errors occurred:

**Comment by 01fiore** [comment]
in etc/luet/repo.conf.d

1) luet.yml
2) mocaccino-extra-stable
3) mocaccino-desktop-stable.yml 
4) mocaccino-os-commons-stable.yml
5) mocaccino-repository-index

# luet install clementine:

Old version                                     | New version                                     | License
apps/flatpak-1.10.2+108                         | apps/flatpak-1.10.2+109                         |        
layers/X-1.0+6                                  | layers/X-1.0+7                                  |        
apps/vim-8.2.3741+35                            | apps/vim-8.2.3741+36                            |        
layers/sys-fs-0.8+9                             | layers/sys-fs-0.8+12                            |        
layers/codecs-6+6                               | layers/codecs-6+9                               |        
apps/firefox-91.6.0+14                          | apps/firefox-91.6.0+15                          |        
apps/mos-updater-0.1+39                         | apps/mos-updater-0.1+40                         |        
apps/vlc-3.0.16+77                              | apps/vlc-3.0.16+78                              |        
layers/system-x-0.20220427+3                    | layers/system-x-0.20220506+1                    |        
apps/grub-2.04+135                              | apps/grub-2.04+136                              |        
layers/qt-5.15.2+145                            | layers/qt-5.15.2+146                            |        
apps/discover-5.22.5+79                         | apps/discover-5.22.5+80                         |        
kernel-modules/nvidia-drivers-lts-470.103.01+19 | kernel-modules/nvidia-drivers-lts-470.103.01+20 |        
layers/kde-apps-minimal-20.12.3+110             | layers/kde-apps-minimal-20.12.3+111             |        
layers/kde-common-4+91                          | layers/kde-common-4+92                          |        
layers/plasma-5.23.5+34                         | layers/plasma-5.23.5+35 

ERROR    Error: Failed matching solutions against repository for clementine where are definitions coming from?!
panic: fatal error

goroutine 1 [running]:
github.com/mudler/luet/pkg/api/core/logger.(*Logger).Fatal(...)
        /luetbuild/luet/pkg/api/core/logger/logger.go:255
github.com/mudler/luet/cmd.glob..func10(0x162a8c0?, {0xc000146c70?, 0x1?, 0x1?})
        /luetbuild/luet/cmd/install.go:94 +0x5c5
github.com/spf13/cobra.(*Command).execute(0x162a8c0, {0xc000146c40, 0x1, 0x1})
        /luetbuild/luet/vendor/github.com/spf13/cobra/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x1628ac0)
        /luetbuild/luet/vendor/github.com/spf13/cobra/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /luetbuild/luet/vendor/github.com/spf13/cobra/command.go:902
github.com/mudler/luet/cmd.Execute()
        /luetbuild/luet/cmd/root.go:124 +0x25
main.main()
        /luetbuild/luet/main.go:10 +0x17

Any suggestion is welcome.

**Reply by mudler** [reply]
Can you try `luet install --relax repository/mocaccino-community` first?

**Reply by msdobrescu** [reply]
Also try `luet uninstall apps/clementine` first.

**Reply by 01fiore** [reply]
I've been using it for a few hours and I didn't know that you had to put "apps".

luet install apps/clementine.

**Comment by msdobrescu** [comment]
Hi, I'm not at the desk now. I don't see community repo in your list above.
Try adding it (again) if missing.

**Reply by 01fiore** [reply]
Solved by reinstalling the repository. Thanks everyone.
